header/header.go

```diff
@@ -223,7 +229,7 @@ func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
 		return err
 	}
 
-	*eh = *out
+	*eh = *out.Clone()
```

Comment:
Why is this necessary?

Reply:
'cause now we've `sync.Once` that cannot (should not) be copied. We've a few places where we do assignment which triggers `go vet`.

So, additional helper method to prevent such things. See failure: https://github.com/celestiaorg/celestia-node/actions/runs/12118920400/job/33784451925?pr=3975#step:4:34

Reply:
Can we find a way to solve this without introducing additional copy?

Reply:
Well, ok, done.